docs/user-guide/src/capm3/remediaton.md

```diff
@@ -0,0 +1,143 @@
+  clusterName: test1
+  # (Optional) maxUnhealthy prevents further remediation if the cluster is already partially unhealthy
+  maxUnhealthy: 100%
+  # (Optional) nodeStartupTimeout determines how long a MachineHealthCheck should wait for
+  # a Node to join the cluster, before considering a Machine unhealthy.
+  # Defaults to 10 minutes if not specified.
+  # Set to 0 to disable the node startup timeout.
+  # Disabling this timeout will prevent a Machine from being considered unhealthy when
+  # the Node it created has not yet registered with the cluster. This can be useful when
+  # Nodes take a long time to start up or when you only want condition based checks for
+  # Machine health.
+  nodeStartupTimeout: 0m
+  # selector is used to determine which Machines should be health checked
+  selector:
+    matchLabels:
+      nodepool: nodepool-0
+  # Conditions to check on Nodes for matched Machines, if any condition is matched for the duration of its timeout, the Machine is considered unhealthy
+  unhealthyConditions:
+  - type: Ready
+    status: Unknown
+    timeout: 300s
+  - type: Ready
+    status: "False"
+    timeout: 300s
+  remediationTemplate: # added infrastructure reference
+    kind: Metal3RemediationTemplate
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    name: worker-remediation-request
+
+```
+
+```Metal3RemediationTemplate``` for worker nodes:
+
+```yaml
+
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3RemediationTemplate
+metadata:
+    name: worker-remediation-request
+    namespace: metal3
+spec:
+  template:
+    spec:
+      strategy:
+        type: "Reboot"
+        retryLimit: 2
+        timeout: 300s
+
+```
+
+### Configuration for controlplane nodes
+
+User can apply the following examples for creating a MachineHealthCheck and Metal3Remediation for controlplane nodes:
+
+```yaml
+
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineHealthCheck
+metadata:
+  name: controlplane-healthcheck
+  namespace: metal3
+spec:
+  clusterName: test1
+  maxUnhealthy: 100%
+  nodeStartupTimeout: 0m
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/control-plane: ""
+  unhealthyConditions:
+    - type: Ready
+      status: Unknown
+      timeout: 300s
+    - type: Ready
+      status: "False"
+      timeout: 300s
+  remediationTemplate: # added infrastructure reference
+    kind: Metal3RemediationTemplate
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    name: controlplane-remediation-request
+
+```
+
+```Metal3RemediationTemplate``` for controlplane nodes:
+
+```yaml
+
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3RemediationTemplate
+metadata:
+    name: controlplane-remediation-request
+    namespace: metal3
+spec:
+  template:
+    spec:
+      strategy:
+        type: "Reboot"
+        retryLimit: 1
+        timeout: 300s
+
+```
+
+## Limitations and Caveats
+
+* Machines owned by a MachineSet or a KubeadmControlPlane can be remediated by a MachineHealthCheck
```

Comment:
Is this a limitation?

Reply:
It can be both limitation and  caveats(warnings). As machine owned by machineset or kcp can be remediated.